client/components/pages/submission-system/register/index.js

```
@@ -19,26 +15,41 @@ export default ({
   authorizedPrograms = [],
   logOut,
   pathname,
-}: any) => (
-  <SubmissionLayout pathname={pathname} logOut={logOut} noSidebar>
-    <div
-      css={css`
-        display: flex;
-        justify-content: center;
-        align-items: center;
-        width: 100%;
-        height: 100%;
-        left: 0px;
-        top: 0px;
-        position: absolute;
-      `}
-    >
-      <JoinProgramForm
-        programName="Pancreatic Cancer - AU"
-        userRole="Program Administrator"
-        onJoinClick={console.log}
-        availableInstitutions={[]}
-      />
-    </div>
-  </SubmissionLayout>
-);
+}: any) => {
+  const { inviteId } = useRouter().query;
+
+  const { data: { joinProgramInvite } = {}, loading, errors } = useQuery(GET_INVITE, {
+    variables: { inviteId },
+  });
+
+  return (
+    <SubmissionLayout pathname={pathname} logOut={logOut} noSidebar>
+      <div
+        css={css`
+          display: flex;
+          justify-content: center;
+          align-items: center;
+          width: 100%;
+          height: 100%;
+          left: 0px;
+          top: 0px;
+          position: absolute;
+        `}
+      >
+        <JoinProgramForm
+          inviteId={inviteId}
+          programName={get(
+            joinProgramInvite,
+            'program.name',
+            loading ? '[loading...]' : '[error to fetch data]',
```

Thank you for handling the error case instead of over looking this. 

Instead of this text which is inconsistent with the rest of the application maybe we should replace the entire form with an error page... If we were unable to fetch user role or program name then we should probably just stop the entire register from taking place until we know what went wrong...

Can you get the values outside of the return statement, and if they are missing (error case) we can replace the entire form with an error screen. Should be asking Kim what this looks like.

@hlminh2000 is that an overkill idea? can you help me simplify this?

---

I think that's fair. if it fails to find the program, something bigger is likely at play so we can probably just throw a 500 error and let the error page handle it. 
Although, if it fails to find the program because of a known business-case (say, program is deleted / invitation expire, etc... ), we can inform user of the reason. That needs to come from spec though

---

That would be an extremely unusual case, i think a generic "somethign went wrong" message would be fine.

The fields are required for the program and user so we shouldn't ever get them back blank without an error on the request (in theory). So i think just a generic "something went wrong" error screen would be fine.

---

then I suggest just throwing a 500 :)

---

use the standard 500 error page?

---

no, actually throwing an error and let Next handle showing the error page :P

---

implying that the gql fetch is done serverside? if thats happening then yeah let next handle it, but it wasn't clear to me that this wasnt loading client side.

---

it could load either side, if we update the `client/pages/_error.js` to show `<Page500 />` as default for unknown error, that should work both sides :)

---

would fix the other error i brought up this morning too

---

Throw 500 based on your discussion.